prow/kube/config.go

```diff
@@ -127,9 +127,6 @@ func mergeConfigs(local *rest.Config, foreign map[string]rest.Config, currentCon
 	if len(ret) == 0 {
 		return nil, errors.New("no client contexts found")
 	}
-	if _, ok := ret[DefaultClusterAlias]; !ok {
-		ret[DefaultClusterAlias] = ret[InClusterContext]
```

Comment:
This seems like it will break other parts of the system (plank, sinker, etc).

Also, the cluster alias "default" will be injected into any job that does not otherwise set this value.